lib/utils/batcher/state.go

```diff
@@ -69,11 +80,20 @@ type StateMonitor struct {
 
 	mu    sync.RWMutex
 	state State
+	// Timer used to auto-exit Overloaded after a quiet period.
+	flushTimer clockwork.Timer
 }
 
 // NewStateMonitor creates a new StateMonitor with the given configuration.
 // The monitor starts in Normal state by default.
 func NewStateMonitor(config StateConfig) *StateMonitor {
+	if config.Clock == nil {
+		config.Clock = clockwork.NewRealClock()
+	}
+	if config.Threshold == 0 {
+		config.Threshold = 100
+	}
```

Comment:
Nit: Can we have `CheckAndSetDefaults()` on the StateConfig to encapsulate this?